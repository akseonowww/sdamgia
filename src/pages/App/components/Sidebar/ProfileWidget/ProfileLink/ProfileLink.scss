.ProfileLink {
  &-Button {
    cursor: pointer;
  }

  &-Icon {
    &:hover {
      opacity: 0.75;
    }
  }
}
