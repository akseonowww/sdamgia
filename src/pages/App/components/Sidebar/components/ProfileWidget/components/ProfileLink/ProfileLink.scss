.ProfileLink {
  &-Icon {
    &:hover {
      opacity: 0.75;
    }
  }
}
