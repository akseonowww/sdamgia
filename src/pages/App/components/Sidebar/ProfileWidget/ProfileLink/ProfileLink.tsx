import React, { FC, useCallback } from 'react'
import { useSelector, useDispatch } from 'react-redux'

import { fetchAuthStatusSuccess } from 'modules/Auth'
import { getAuthUser } from 'modules/Auth/selectors'
import Link from 'components/Link'
import './ProfileLink.scss'

import signOutIcon from './assets/signOutIcon.png'

const ProfileLink: FC = () => {
  const user = useSelector(getAuthUser)
  const dispatch = useDispatch()

  const signOut = useCallback(
    (e) => {
      e.preventDefault()
      dispatch(fetchAuthStatusSuccess(false))
    },
    [dispatch]
  )

  return (
    <>
      <div className="ProfileLink">
        <Link className="Link_static ProfileWidget-Link" href="/profile">
          {user && user.name}
        </Link>
      </div>
      <a href="/" onClick={(e) => signOut(e)}>
        <img className="ProfileLink-Icon" src={signOutIcon} alt="Выйти" />
      </a>
    </>
  )
}

export default ProfileLink
