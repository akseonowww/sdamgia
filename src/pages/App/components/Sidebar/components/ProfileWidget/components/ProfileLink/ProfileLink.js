import React, { useCallback } from 'react';

import './ProfileLink.scss';

import signOutIcon from './assets/signOutIcon.png';

const ProfileLink = ({ setAuth }) => {
	const signOut = useCallback(
		e => {
			e.preventDefault();
			setAuth(false);
		},
		[setAuth]
	);

	return (
		<>
			<div className="">
				<a
					className="Link Link_static ProfileWidget-Link"
					href="/profile"
				>
					Константин
				</a>
			</div>
			<a href="/" onClick={e => signOut(e)}>
				<img
					className="ProfileLink-Icon"
					src={signOutIcon}
					alt="Выйти"
				/>
			</a>
		</>
	);
};

export default ProfileLink;
