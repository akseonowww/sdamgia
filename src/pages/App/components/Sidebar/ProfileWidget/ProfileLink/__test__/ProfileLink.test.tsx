import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import ProfileLink from '..'

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn().mockReturnValueOnce(jest.fn()),
}))
jest.mock('modules/Auth/selectors', () => ({
  getAuthUser: jest.fn().mockReturnValueOnce(''),
}))

describe('ProfileLink', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders stub when no user name', () => {
    act(() => {
      render(<ProfileLink />, container)
    })
    expect(container && container.textContent).toBe('Профиль')
  })

  it('matches snapshot', () => {
    act(() => {
      render(<ProfileLink />, container)
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
