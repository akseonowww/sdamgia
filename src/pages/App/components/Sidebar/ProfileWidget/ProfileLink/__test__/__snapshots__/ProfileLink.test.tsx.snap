// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileLink matches snapshot 1`] = `"<div class=\\"ProfileLink\\"><a class=\\"Link Link_static ProfileWidget-Link\\" href=\\"/profile\\" rel=\\"noopener noreferrer\\">Профиль</a></div><a href=\\"/\\"><img class=\\"ProfileLink-Icon\\" src=\\"signOutIcon.png\\" alt=\\"Выйти\\"></a>"`;
